store/postgres/src/relational_queries.rs

```diff
@@ -338,31 +338,26 @@ impl<'a> QueryFragment<Pg> for QueryValue<'a> {
             },
             Value::Int(i) => out.push_bind_param::<Integer, _>(i),
             Value::BigDecimal(d) => {
-                // If a BigDecimal with negative scale gets sent to Diesel (and then Postgres)
-                // Postgres will complain with 'invalid scale in external "numeric" value'
-                let (_, scale) = d.as_bigint_and_exponent();
-                if scale < 0 {
-                    // This is safe since we are effectively multiplying the
-                    // integer part of d with 10^(-scale), an integer
-                    let d = d.with_scale(0);
-                    out.push_bind_param::<Numeric, _>(&d)
-                } else {
-                    out.push_bind_param::<Numeric, _>(d)
-                }
+                out.push_bind_param::<Text, _>(&d.to_string())?;
+                out.push_sql("::numeric");
+                Ok(())
             }
             Value::Bool(b) => out.push_bind_param::<Bool, _>(b),
             Value::List(values) => {
-                let values = SqlValue::new_array(values.clone());
+                let sql_values = SqlValue::new_array(values.clone());
```

Comment:
Minor quibble about the variable renaming: I usually rebind the same variable in situations like this to indicate to the reader that the old value will not be used in the code that follows. That reduces the number of things to think about, and helps making things more understandable.

Reply:
Nothing against that, it's just that I actually needed the old value in this case, and couldn't come up with something more elegant.

Reply:
Oh, woops .. I'll get off my soapbox then ;)